pkg/api/vm/handler.go

```diff
@@ -622,25 +622,75 @@ func isDrained(node *corev1.Node) bool {
 }
 
 func (h *vmActionHandler) getNodeSelectorRequirementFromVMI(vmi *kubevirtv1.VirtualMachineInstance) (labels.Selector, error) {
-	if vmi == nil || vmi.Spec.Affinity == nil || vmi.Spec.Affinity.NodeAffinity == nil || vmi.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
-		return labels.Everything(), nil
+	nodeSelector := labels.NewSelector()
+
+	if vmi != nil && vmi.Spec.Affinity != nil && vmi.Spec.Affinity.NodeAffinity != nil && vmi.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
```

Comment:
The CPU model and feature labels are added to the `virt-launcher` pod's `.spec.nodeSelector`, not the VMI's node affinity. Ultimately, it's the pod that gets scheduled to the node, not the VMI.